File: src/BaseHmiTypes/Cycles/HmiCycleList.cs
namespace BaseHmiTypes.Cycles;

public sealed class HmiCycleList : IHmiObject
{
    public DateTime? LastModified { get; set; }

    public string? Name { get; set; }

    public IList<HmiCycle> HmiCycles { get; } = new List<HmiCycle>();
}

File: typescript/base-hmi-types/src/cycles/HmiCycleList.ts
import { IHmiObject } from "../IHmiObject.js";
import { HmiCycle } from "./HmiCycle.js";

export class HmiCycleList implements IHmiObject {
  lastModified?: Date;
  name?: string;
  readonly hmiCycles: HmiCycle[] = [];
}
